include/aos/common/cloudprotocol/certificates.hpp

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_CLOUDPROTOCOL_CERTIFICATES_HPP_
+#define AOS_CLOUDPROTOCOL_CERTIFICATES_HPP_
+
+#include "aos/common/cloudprotocol/cloudprotocol.hpp"
+#include "aos/common/crypto/crypto.hpp"
+#include "aos/common/tools/map.hpp"
+#include "aos/common/tools/optional.hpp"
+#include "aos/common/types.hpp"
+
+namespace aos::cloudprotocol {
+
+/**
+ * Supported version of UnitSecret message.
+ */
+static constexpr auto cUnitSecretVersion = "2.0.0";
+
+/**
+ * Certificate installation description size.
+ */
+static constexpr auto cCertInstallationDescriptionSize
+    = AOS_CONFIG_CLOUDPROTOCOL_CERTIFICATE_INSTALLATION_DESCRIPTION_SIZE;
+
+/**
+ * Certificate secret size.
+ */
+static constexpr auto cCertSecretSize = AOS_CONFIG_CLOUDPROTOCOL_CERTIFICATE_SECRET_SIZE;
+
+/**
+ * Maximum number of certificates per node.
+ */
+static constexpr auto cCertificatesPerNodeCount = AOS_CONFIG_CLOUDPROTOCOL_CERTIFICATES_PER_NODE_COUNT;
+
+/**
+ * Maximum number of certificates per unit.
+ */
+static constexpr auto cCertsPerUnitCount = cMaxNumNodes * cCertificatesPerNodeCount;
+
+/**
+ * Cert type type.
+ */
+class CertTypeType {
+public:
+    enum class Enum {
+        eOffline,
+        eOnline,
+        eUM,
+        eSM,
+        eCM,
+        eIAM,
+        eHostDevice,
+        eRemoteDevice,
+        eAzureIot,
+    };
+
+    static const Array<const char* const> GetStrings()
+    {
+        static const char* const sStrings[] = {
+            "offline",
+            "online",
+            "um",
+            "sm",
+            "cm",
+            "iam",
+            "host-device",
```

Comment:
Not sure we use host-device, remote-device and azure-iot. Need to check with cloud team.